drivers/i2c/i2c_mcux_lpi2c.c

```diff
@@ -297,12 +305,175 @@ static int mcux_lpi2c_recover_bus(const struct device *dev)
+			if (target_cb->write_requested) {
+				ret = target_cb->write_requested(data->target_cfg);
+				if (ret < 0) {
+					/* NAK further bytes */
+					data->send_ack = false;
+				}
+			}
+		}
+		if (target_cb->write_received) {
+			ret = target_cb->write_received(data->target_cfg,
+							i2c_data);
+			if (ret < 0) {
+				/* NAK further bytes */
+				data->send_ack = false;
+			}
+		}
+	}
+	if (flags & kLPI2C_SlaveTxReadyFlag) {
+		/* Space is available in TX fifo, issue callback and write out */
+		if (data->first_tx) {
+			data->read_active = true;
+			data->first_tx = false;
+			if (target_cb->read_requested) {
+				ret = target_cb->read_requested(data->target_cfg,
+								&i2c_data);
+				if (ret < 0) {
+					/* Disable TX */
+					data->read_active = false;
+				} else {
+					/* Send I2C data */
+					config->base->STDR = i2c_data;
+				}
+			}
+		} else if (data->read_active) {
+			if (target_cb->read_processed) {
+				ret = target_cb->read_processed(data->target_cfg,
+								&i2c_data);
+				if (ret < 0) {
+					/* Disable TX */
+					data->read_active = false;
+				} else {
+					/* Send I2C data */
+					config->base->STDR = i2c_data;
+				}
+			}
+		}
+	}
+	if (flags & kLPI2C_SlaveStopDetectFlag) {
+		LPI2C_SlaveClearStatusFlags(config->base, flags);
+		if (target_cb->stop) {
+			target_cb->stop(data->target_cfg);
+		}
+	}
+	if (flags & kLPI2C_SlaveTransmitAckFlag) {
+		LPI2C_SlaveTransmitAck(config->base, data->send_ack);
+	}
+}
+
+static int mcux_lpi2c_target_register(const struct device *dev,
+			struct i2c_target_config *target_config)
+{
+	const struct mcux_lpi2c_config *config = dev->config;
+	struct mcux_lpi2c_data *data = dev->data;
+	lpi2c_slave_config_t slave_config;
+	uint32_t clock_freq;
+
+	LPI2C_MasterDeinit(config->base);
+
+	/* Get the clock frequency */
+	if (clock_control_get_rate(config->clock_dev, config->clock_subsys,
+				   &clock_freq)) {
+		return -EINVAL;
+	}
+
+	if (!target_config) {
+		return -EINVAL;
+	}
+
+	if (data->target_attached) {
+		return -EBUSY;
+	}
+
+	data->target_attached = true;
+	data->target_cfg = target_config;
+	data->first_tx = false;
+
+	LPI2C_SlaveGetDefaultConfig(&slave_config);
+	slave_config.address0 = target_config->address;
+	/* Note- this setting enables clock stretching to allow the
+	 * slave to respond to each byte with an ACK/NAK.
+	 * this behavior may cause issues with some I2C controllers.
+	 */
+	slave_config.sclStall.enableAck = true;
+	LPI2C_SlaveInit(config->base, &slave_config, clock_freq);
+	/* Clear all flags. */
+	LPI2C_SlaveClearStatusFlags(config->base, (uint32_t)kLPI2C_SlaveClearFlags);
+	/* Enable interrupt */
+	LPI2C_SlaveEnableInterrupts(config->base,
+					(kLPI2C_SlaveTxReadyFlag |
+					kLPI2C_SlaveRxReadyFlag |
+					kLPI2C_SlaveStopDetectFlag |
+					kLPI2C_SlaveAddressValidFlag |
+					kLPI2C_SlaveTransmitAckFlag));
+	return 0;
+}
+
+static int mcux_lpi2c_target_unregister(const struct device *dev,
+			       struct i2c_target_config *target_config)
```

Comment:
Indentation.

Reply:
I have no problem adjusting the indentation here, but what are you looking for? Alignment with the prior `(`? I don't see any specific guidelines on how to indent the second line of a function declaration within Zephyr.